WooCommerce/Classes/Authentication/Epilogue/StorePickerCoordinator.swift

```diff
@@ -34,6 +34,26 @@ final class StorePickerCoordinator: Coordinator {
     func start() {
         showStorePicker()
     }
+
+    /// A static method which allows easily to switch store
+    ///
+    static func switchStore(with storeID: Int64, onCompletion: @escaping SelectStoreClosure) {
+        guard storeID != ServiceLocator.stores.sessionManager.defaultStoreID else {
+            onCompletion()
+            return
+        }
+
+        let storePicker = StorePickerCoordinator(UINavigationController(), config: .switchingStores)
```

Comment:
`StorePickerCoordinator` seems to be doing too many things. It switches updates `ServiceLocator.stores` **and** shows a UI too. This is why we have this weird `UINavigationController()` argument here. 

I think in the future we should extract the business logic here into a `SwitchStoreUseCase` and then that one is shared by the `StorePickerCoordinator` and `MainTabBarController`.

![image](https://user-images.githubusercontent.com/198826/81086723-1c682200-8eb6-11ea-8c10-c560472afa7c.png)

For now, I think we should not encourage usage of this `static` function by other classes in the future. Would you consider making this a `private func` inside `MainTabBarController` for now?

Reply:
Sorry for the delay @shiki in responding to your comment. I agree with you, and I like your proposal, the `StorePickerCoordinator` does too many things.

I implemented directly your proposed solution since it doesn't make much sense to me to move the switch method under the `MainTabBarController`. Why postpone? 🛩️ 

I moved some methods, in particular, the ones that update the `ServiceLocator.stores` under `SwitchStoreUseCase`. I set up the class in way where it's possible to injected via the constructor the two `ServiceLocator` that we are using. This can be useful in the future for writing some unit tests because, at the moment, all the tests override the app singletons in ServiceLocator, and it could lead to tricky issues and unexpected setup. It's the same problem (maybe it's better to call it tech debt) already reported by @jaclync here: https://github.com/woocommerce/woocommerce-ios/issues/1909

Would you like to take a second look at it?
Things to be tested in addition to the first one:
- Login flow
- Switch store from app settings

🙇‍♂️

Reply:
Taking a look today!